### DISCUSSION TITLE
Could This Reasonably Be Used to Build a Code Formatter?

### DISCUSSION THREAD

**Discussion by zicklag** [egraphs-good/egg#93] [General]
I was thinking about potential use-cases for `egg` and I was wondering whether or not it makes sense to you to make a code formatter based on it.

It seems like it's kind of similar to an optimization problem: you're taking input tokens and generating replacements for those tokens, but instead of optimizing for AST-size, or operation expense, etc., you are optimizing for "prettiness".

Does that sound reasonable to you who actually know stuff about EGraphs, or is it maybe not the best use-case?

**Comment by mwillsey** [comment]
Hmm, I think it _technically_ could be, but that doesn't strike me as a good use case for e-graphs. In code formatting, it's more about canonicalization rather than optimization: you know exactly how to format a given set of tokens, you don't need to search among many possibilities.

**Reply by zicklag** [reply]
Yeah, after thinking about it for a while I couldn't think of a super great way to use it and it seems like it might not be the best use-case. Thanks for the reply. :slightly_smiling_face:
